guacamole-docker/bin/start.sh

```diff
@@ -708,6 +708,58 @@ associate_json() {
     # Add required .jar files to GUACAMOLE_EXT
     ln -s /opt/guacamole/json/guacamole-auth-*.jar "$GUACAMOLE_EXT"
 }
+##
+## Sets up Tomcat's remote IP valve that allows gathering the remote IP
+## from headers set by a remote proxy
+## Upstream documentation: https://tomcat.apache.org/tomcat-8.5-doc/api/org/apache/catalina/valves/RemoteIpValve.html
+##
+enable_remote_ip_valve() {
+    # Add <Valve> element
+    xmlstarlet edit --inplace \
+        --insert '/Server/Service/Engine/Host/*' --type elem -n Valve \
+        --insert '/Server/Service/Engine/Host/Valve[not(@className)]' --type attr -n className -v org.apache.catalina.valves.RemoteIpValve \
+        $CATALINA_BASE/conf/server.xml
+
+    # Allowed IPs
+    if [ -z "$GUACAMOLE_PROXY_ALLOWED_IPS_REGEX" ]; then
```

Comment:
`GUACAMOLE_PROXY_ALLOWED_IPS_REGEX`, `GUACAMOLE_PROXY_IP_HEADER`, etc. feel redundant with their `GUACAMOLE_` prefix, which is not used by any of the other Docker-specific environment variables for the `guacamole/guacamole` image. It might also result in some confusion, as guacd is sometimes referred to as the Guacamole proxy daemon.